sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,401 @@
+            "collection_interval": collection_interval,
+            "dbms_version": None,
+            "tags": self._check.non_internal_tags,
+            "cloud_metadata": self._check._config.cloud_metadata,
+        }
+        self._data_submitter = SubmitData(self._check.database_monitoring_metadata, base_event, self._log)
+
+    def run_job(self):
+        self._collect_schemas_data()
+
+    def shut_down(self):
+        self._data_submitter.submit()
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _collect_schemas_data(self):
+        """Collects database information and schemas and submits to the agent's queue as dictionaries
+        schema dict
+        key/value:
+            "name": str
+            "id": str
+            "owner_name": str
+            "tables" : list of tables dicts
+                table
+                key/value:
+                    "id" : str
+                    "name" : str
+                    columns: list of columns dicts
+                        columns
+                        key/value:
+                            "name": str
+                            "data_type": str
+                            "default": str
+                            "nullable": bool
+                indexes : list of index dicts
+                    index
+                    key/value:
+                        "name": str
+                        "type": str
+                        "is_unique": bool
+                        "is_primary_key": bool
+                        "is_unique_constraint": bool
+                        "is_disabled": bool,
+                        "column_names": str
+                foreign_keys : list of foreign key dicts
+                    foreign_key
+                    key/value:
+                        "foreign_key_name": str
+                        "referencing_table": str
+                        "referencing_column": str
+                        "referenced_table": str
+                        "referenced_column": str
+                partitions: partition dict
+                    partition
+                    key/value:
+                        "partition_count": int
+        """
+        start_time = time.thread_time()
+        self._data_submitter.reset()
+        self._data_submitter.set_base_event_data(
+            self._check.resolved_hostname,
+            self._check.non_internal_tags,
+            self._check._config.cloud_metadata,
+            "{},{}".format(
+                self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+                self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            ),
+        )
+
+        databases = self._check.get_databases()
+        db_infos = self._query_db_informations(databases)
+        self._data_submitter.store_db_infos(db_infos)
+
+        @tracked_method(agent_check_getter=agent_check_getter)
+        def fetch_schema_data(cursor, db_name):
```

Comment:
Why do you choose define `fetch_schema_data` inside `_collect_schemas_data`? can `fetch_schema_data` be defined outside of `_collect_schemas_data`, possibly as a private function of the class?

Reply:
I think I am used to define "lambdas" from C++ in that way, no particular reason otherwise. I'll modify.